antarest/study/service.py

```diff
@@ -2384,6 +2385,15 @@ def upgrade_study(
         assert_permission(params.user, study, StudyPermissionType.WRITE)
         self._assert_study_unarchived(study)
 
+        # TODO : This catch is temporary and should be removed once the commands upgrade is fully implemented
```

Comment:
Pour info, si on utilise un "TODO", cela ajoute un warning dans SonarCloud, chose que je cherche à éviter.

En fait, c'est pas vraiment un TODO, car on n'a pas les spécifications utilisateurs. Donc, on ne sais même pas si on va implémenter cela.

Par contre, on peut mettre un commentaire explicatif, par exemple :

The upgrade of a study variant requires the use of a command specifically dedicated to the upgrade. However, such a command does not currently exist. Moreover, upgrading a study (whether raw or variant) directly impacts its descendants, as it would necessitate upgrading all of them. It’s uncertain whether this would be an acceptable behavior. For this reason, upgrading a study is not possible if the study is a variant or if it has descendants.